internal/manifests/targetallocator/targetallocator.go

```diff
@@ -46,6 +47,14 @@ func Build(params Params) ([]client.Object, error) {
 		resourceFactories = append(resourceFactories, manifests.FactoryWithoutError(ServiceMonitor))
 	}
 
+	if params.Config.CertManagerAvailability() == certmanager.Available && featuregate.EnableTargetAllocatorMTLS.IsEnabled() {
+		resourceFactories = append(resourceFactories, manifests.FactoryWithoutError(SelfSignedIssuer))
```

Comment:
you should be able to do a single append here `append(resourceFactories, manifests.FactoryWithoutError(SelfSignedIssuer), manifests.FactoryWithoutError(CACertificate), ...` etc.